core/src/test/scala/unit/kafka/admin/AdminRackAwareTest.scala

```diff
@@ -0,0 +1,226 @@
+    val newList = AdminUtils.getRackAlternatedBrokerList(rackMap)
+    assertEquals(List(0, 3, 1, 5, 4, 2), newList)
+    val anotherList = AdminUtils.getRackAlternatedBrokerList(rackMap - 5)
+    assertEquals(List(0, 3, 1, 4, 2), anotherList)
+    val assignment = AdminUtils.assignReplicasToBrokers(toBrokerMetadata(rackMap), 7, 3, 0, 0)
+    val expected = Map(0 -> List(0, 3, 1),
+                       1 -> List(3, 1, 5),
+                       2 -> List(1, 5, 4),
+                       3 -> List(5, 4, 2),
+                       4 -> List(4, 2, 0),
+                       5 -> List(2, 0, 3),
+                       6 -> List(0, 4, 2))
+    assertEquals(expected, assignment)
+  }
+
+  @Test
+  def testAssignmentWithRackAware() {
+    val brokerRackMapping = Map(0 -> "rack1", 1 -> "rack2", 2 -> "rack2", 3 -> "rack3", 4 -> "rack3", 5 -> "rack1")
+    val numPartitions = 6
+    val replicationFactor = 3
+    val assignment = AdminUtils.assignReplicasToBrokers(toBrokerMetadata(brokerRackMapping), numPartitions,
+      replicationFactor, 2, 0)
+    ensureRackAwareAndEvenDistribution(assignment, brokerRackMapping, brokerRackMapping.size, numPartitions,
+      replicationFactor)
+  }
+
+  @Test
+  def testAssignmentWithRackAwareWithRandomStartIndex() {
+    val brokerRackMapping = Map(0 -> "rack1", 1 -> "rack2", 2 -> "rack2", 3 -> "rack3", 4 -> "rack3", 5 -> "rack1")
+    val numPartitions = 6
+    val replicationFactor = 3
+    val assignment = AdminUtils.assignReplicasToBrokers(toBrokerMetadata(brokerRackMapping), numPartitions,
+      replicationFactor)
+    ensureRackAwareAndEvenDistribution(assignment, brokerRackMapping, brokerRackMapping.size, numPartitions,
+      replicationFactor)
+  }
+
+  @Test
+  def testAssignmentWithRackAwareWithNotEnoughPartitions() {
+    val brokerRackMapping = Map(0 -> "rack1", 1 -> "rack2", 2 -> "rack2", 3 -> "rack3", 4 -> "rack3", 5 -> "rack1")
+    val numPartitions = 13
+    val replicationFactor = 3
+    val assignment = AdminUtils.assignReplicasToBrokers(toBrokerMetadata(brokerRackMapping), numPartitions,
+      replicationFactor, 0, 0)
+    ensureRackAwareAndEvenDistribution(assignment, brokerRackMapping, brokerRackMapping.size, numPartitions,
+      replicationFactor, verifyLeaderDistribution = false, verifyReplicasDistribution = false)
+  }
+
+  @Test
+  def testAssignmentWithRackAwareWithUnevenRacks() {
+    val brokerRackMapping = Map(0 -> "rack1", 1 -> "rack1", 2 -> "rack2", 3 -> "rack3", 4 -> "rack3", 5 -> "rack1")
+    val numPartitions = 12
+    val replicationFactor = 3
+    val assignment = AdminUtils.assignReplicasToBrokers(toBrokerMetadata(brokerRackMapping), numPartitions,
+      replicationFactor)
+    ensureRackAwareAndEvenDistribution(assignment, brokerRackMapping, brokerRackMapping.size, numPartitions,
+      replicationFactor, verifyReplicasDistribution = false)
+  }
+
+
+  @Test
+  def testAssignmentWithRackAwareWith12Partitions() {
```

Comment:
Probably not. :)

In general, these tests run very fast since all they do is operate on collections in memory. So I have not thought about reducing the number of tests.